### ISSUE TITLE
[BUG]: DeepFace verfiy function tends to necessarily find a face for verification

### ISSUE THREAD

**Issue by JohannesRie** [serengil/deepface#1342]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

v0.0.90

### Python version

3.9.19

### Operating System

Windows 11

### Dependencies

tensorboard-data-server==0.7.2
tensorboard-plugin-wit==1.8.1
tensorboardX==2.6.2.2
tensorflow==2.13.1
tensorflow-estimator==2.13.0
tensorflow-intel==2.13.1
tensorflow-io-gcs-filesystem==0.31.0
keras==2.13.1
keras-nightly==2.5.0.dev2021032900
Keras-Preprocessing==1.1.2
pillow==10.4.0

### Reproducible example

```Python
(...)
try:
    temp_img = Image.open(rec_pic)     # rec_pic is file-like object
    temp_rec_pic_path = f"temp/"+rec_pic.name
    with open(temp_rec_pic_path, "w") as f1:
        temp_img.save(f1)
        f1.close()

     temp_img = Image.open(comp_pic)     # comp_pic is file-like object
     temp_comp_pic_path = f"temp/"+comp_pic.name
     with open(temp_comp_pic_path, "w") as f2:
        temp_img.save(f2)
        f2.close()
     
    obj = DeepFace.verify(str(temp_rec_pic_path), 
                          str(temp_comp_pic_path), 
                          model_name = 'Facenet512',  
                          detector_backend = 'retinaface', 
                          align = True)
(...)
```

### Expected Result

Out of interest, I created a relatively large test data set with approx. 2000 images in which prominent people are depicted. I paid attention to various attributes of the people (gender, age, skin color ...) and also made sure to include visual doubles of the celebrities (e.g. Mark Strong and Stanley Tucci, Henry Cavill and Matt Bomer ...). Expectation: The software can tell similar people apart.

### What happened instead?

I performed extensive comparisons with DeepFace verify and noticed that DeepFace tends to recognize a person from the reference photo in each image. Regardless of whether the reference person is included in the photo.  Admittedly, the cosine distance was in an high range (median 0,589) and was absolutely not comparable with the cosine distances of the correct recognitions (median 0,357). However, it was still remarkable.

### Additional Info

Awesome project :-) Thank you!

**Comment by serengil** [comment]
As mentioned in [its documentation](https://github.com/serengil/deepface/tree/master/benchmarks), FaceNet512d offers 98.4% accuracy on LFW dataset. Additionally, no ML model can offer 100% accuracy. 

Closing this due to it is not a bug.

**Comment by serengil** [comment]
Okay as I mentioned ML models come with true positive misclassifications. Nothing to do for that case.
